tests/RPCServer.test.ts

```diff
@@ -641,120 +641,120 @@ describe(`${RPCServer.name}`, () => {
-  //         transform: (chunk, controller) => {
-  //           if ('result' in chunk) chunk.result = { value: 1 };
-  //           controller.enqueue(chunk);
-  //         },
-  //       }),
-  //     };
-  //   });
-  //   const rpcServer = new RPCServer({
-  //     middlewareFactory: middleware,
-  //     logger,
-  //     idGen,
-  //   });
-  //   await rpcServer.start({
-  //     manifest: {
-  //       testMethod: new TestMethod({}),
-  //     },
-  //   });
-  //   const [outputResult, outputStream] = rpcTestUtils.streamToArray();
-  //   const readWriteStream: RPCStream<Uint8Array, Uint8Array> = {
-  //     cancel: () => {},
-  //     readable: stream,
-  //     writable: outputStream,
-  //   };
-  //   rpcServer.handleStream(readWriteStream);
-  //   const out = await outputResult;
-  //   expect(out.map((v) => v!.toString())).toStrictEqual(
-  //     messages.map(() =>
-  //       JSON.stringify({
-  //         jsonrpc: '2.0',
-  //         result: { value: 1 },
-  //         id: null,
-  //       }),
-  //     ),
-  //   );
-  //   await rpcServer.stop({ force: true });
-  // });
+  test.prop({
+    messages: specificMessageArb,
+  })('forward middlewares', async ({ messages }) => {
+    const stream = rpcTestUtils.messagesToReadableStream(messages);
+    class TestMethod extends DuplexHandler {
+      public handle = async function* (
+        input: AsyncGenerator<JSONRPCRequestParams>,
+        _cancel: (reason?: any) => void,
+        _meta: Record<string, JSONValue> | undefined,
+        _ctx: ContextTimed,
+      ): AsyncGenerator<JSONRPCResponseResult> {
+        yield* input;
+      };
+    }
+    const middlewareFactory = rpcUtilsMiddleware.defaultServerMiddlewareWrapper(
+      () => {
+        return {
+          forward: new TransformStream({
+            transform: (chunk, controller) => {
+              chunk.params = { value: 1 };
+              controller.enqueue(chunk);
+            },
+          }),
+          reverse: new TransformStream(),
+        };
+      },
+    );
+    const rpcServer = new RPCServer({
+      middlewareFactory: middlewareFactory,
+      logger,
+      idGen,
+    });
+    await rpcServer.start({
+      manifest: {
+        testMethod: new TestMethod({}),
+      },
+    });
+    const [outputResult, outputStream] = rpcTestUtils.streamToArray();
+    const readWriteStream: RPCStream<Uint8Array, Uint8Array> = {
+      cancel: () => {},
+      readable: stream,
+      writable: outputStream,
+    };
+    rpcServer.handleStream(readWriteStream);
+    const out = await outputResult;
+    expect(out.map((v) => v!.toString())).toStrictEqual(
+      messages.map(() =>
+        JSON.stringify({
+          jsonrpc: '2.0',
+          result: { value: 1 },
+          id: null,
+        }),
+      ),
+    );
+    await rpcServer.stop({ force: true });
+  });
+  test.prop(
+    {
+      messages: specificMessageArb,
+    },
+    // { numRuns: 1 },
+    { seed: 1292472631, path: "0", endOnFailure: true }
```

Comment:
I copied this failing run from a previously failing action, but now, all the tests are passing without failures. This was also the case before, where all the tests passed during development in my branch, and locally too, but when I merged the branch, the tests began failing everywhere.

I tried to uncomment the tests and add a known failing case to replicate the issue so I can attempt to fix it, but I couldn't do that, as the tests are now all passing.

Action: https://github.com/MatrixAI/js-rpc/actions/runs/11733584104/job/32688941570#step:5:269

Reply:
I might try modifying the CI file to run CI across all platforms like MacOS and Windows, just in case the issue is triggered via a run on MacOS.

Reply:
Well, I didn't manage to update the CI file, so I just updated the actual code to now avoid this issue. Hopefully updating the way streams are handled should allow us to avoid this issue entirely.